flocker/node/_deploy.py

```diff
@@ -320,6 +327,36 @@ def run(self, deployer):
 
 
 @implementer(IStateChange)
+class DeleteDataset(PRecord):
+    """
+    Delete all local copies of the dataset.
+
+    A better action would be one that deletes a specific manifestation
+    ("volume" in flocker.volume legacy terminology). Unfortunately
+    currently "remotely owned volumes" (legacy terminology), aka
+    non-primary manifestations or replicas, are not exposed to the
+    deployer, so we have to enumerate them here.
+
+    :ivar Dataset: The dataset to delete.
+    """
+    dataset = field(mandatory=True, type=Dataset)
+
+    def run(self, deployer):
+        service = deployer.volume_service
+        d = service.enumerate()
+
+        def got_volumes(volumes):
+            deletions = []
+            for volume in volumes:
+                if volume.name.dataset_id == self.dataset.dataset_id:
+                    deletions.append(service.pool.destroy(volume).addErrback(
+                        write_failure, _logger, u"flocker:p2pdeployer:delete"))
```

Comment:
I think we really need to document what we log.

Can we start a reference document that gives an exhaustive list of "systems" and what they indicate?  Or should people just refer to the implementation when they want to know what a particular log event represents?